### PR TITLE
NHWC training support

### PR DESCRIPTION
**Pull request by YangFei1990**
1. Add NHWC support for all conv layers
2. Add support for TF1.1.4 by add custom transpose conv from [tensorpack](https://github.com/tensorpack/tensorpack/blob/7e923bfe7fdb97f5e5f9c9ef0df9967658fa5b5f/tensorpack/models/conv2d.py#L204)
3. Some minor changes, including disable autograph for batch norm and changing default training set to coco2017

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tensorpack/models/conv2d.py [hunk 1]
```diff
@@ -167,27 +208,79 @@ def Conv2DTranspose(
...
+                bias_regularizer=bias_regularizer,
+                activity_regularizer=activity_regularizer,
+                _reuse=tf.get_variable_scope().reuse)
+            ret = layer.apply(inputs, scope=tf.get_variable_scope())
+            ret = tf.identity(ret, name='output')
+        ret.variables = VariableHolder(W=layer.kernel)
+        if use_bias:
+            ret.variables.b = layer.bias
+    else:
+        # Our own implementation, to avoid Keras bugs. https://github.com/tensorflow/tensorflow/issues/25946
```
Comment (pidajay) [inline tensorpack/models/conv2d.py]: This comment says 'our own'. I assume it means tensorpack? Maybe you can change the wording

File: infra/docker/train_multinode.sh [hunk 2]
```diff
@@ -43,11 +50,17 @@ DATA.TRAIN='["train2017"]' \
...
+TRAIN.FPN_NCHW=False \
+TRAIN.RPN_NCHW=False \
+TRAIN.MASK_NCHW=False \
+TRAIN.WARMUP_INIT_LR=0.000416666666667 \
+FRCNN.BBOX_REG_WEIGHTS='[20., 20., 10., 10.]' \
+TRAIN.GRADIENT_CLIP=0 \
 TRAINER=horovod
-#For 32x4
-#TRAIN.GRADIENT_CLIP=1.5
+# set to 1.5 to enable clip
```
Comment (pidajay) [inline infra/docker/train_multinode.sh]: Maybe extend comment here saying loss can be NaN if gradient clip is not set for large batch sizes

File: tensorpack/models/conv2d.py (hunk 1 above)
Comment (YangFei1990) [reply tensorpack/models/conv2d.py]: Done

File: infra/docker/train_multinode.sh (hunk 2 above)
Comment (YangFei1990) [reply infra/docker/train_multinode.sh]: Done

File: MaskRCNN/config.py [hunk 3]
```diff
@@ -133,6 +133,10 @@ def __ne__(self, _):
 _C.TRAIN.BATCH_SIZE_PER_GPU = 1
 _C.TRAIN.SEED = 1234
 _C.TRAIN.GRADIENT_CLIP = 0 # set non-zero value to enable gradient clip, 0.36 is recommended for 32x4
```
Comment (johnbensnyder) [inline MaskRCNN/config.py]: Do we want TRAIN.GRADIENT_CLIP to be on or off by default?

Comment (YangFei1990) [reply MaskRCNN/config.py]: I think so. Customer (or someone else who use the repo) usually don't have too many nodes, it is not likely that they will hit the NaN problem.

**Comment by johnbensnyder** [comment]
LGTM
